### DISCUSSION TITLE
Time Complexity Estimate

### DISCUSSION THREAD

**Discussion by pikaduck** [milvus-io/milvus#22727] [Q&A and General discussion]
Could the authors add some literature on the way hybrid search works in terms of its time complexity & how it sees items in the database, whether it compares with all items or not, and other points like this which can help understand the way search works and optimise it for specific use cases

**Comment by yhmo** [comment]
This article might can answer your question: https://milvus.io/blog/2022-2-14-bitset.md

**Comment by xiaofan-luan** [comment]
> Could the authors add some literature on the way hybrid search works in terms of its time complexity & how it sees items in the database, whether it compares with all items or not, and other points like this which can help understand the way search works and optimise it for specific use cases

Hi  pikuduck~
there are acually two mode, brute force and indexed. 
For brute force every entity has to be checked. so it's O(N)
For Index, it's like traditional inverted index and it's O(1)
